### PR TITLE
new command for running 'git log -S' against all repos in the team

### PR DESCRIPTION
**Pull request by andy-duncan**
@glasseyes42 I know this is not exactly what you are looking for, but might still be quite useful in a number of instances. Maybe we should have a team that has ALL the repos in it?  ;-)

not totally convinced that 'history' is the right name for it, but I thought it was a start in the right direction.

### CODE REVIEW

**Comment by cliftonc** [comment]
@andyduncan how is this different to the current `activity` command?  

Also, we use to have all the repos against 'staff', I had a script that made sure they were all part of that team - let me find it.

**Comment by andy-duncan** [comment]
I think that 'activity' will just show recent.

The 'git log -S<search term>' will show any commits in the history that contain the search term. Useful for trying to find if a usage of a variable was removed (not evident in a blame/annotate).

**Comment by cliftonc** [comment]
Got it.

File: commands/history.js [hunk 1]
```diff
@@ -0,0 +1,61 @@
+var async = require('async');
+var execFile = require('child_process').execFile;
+
+module.exports = {
+  name: 'grep',
```
Comment (geophree) [inline commands/history.js]: This should be `'history'`

File: commands/history.js [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+  description: 'runs git log -S across your repos, use -- to separate bosco options from git grep options',
```
Comment (geophree) [inline commands/history.js]: Briefly explain what `git log -S` does in the description, please.

Comment (andy-duncan) [inline commands/history.js]: thanks, Jeff. I think this was just hacked together in a few minutes to demonstrate to @glasseyes42 that maybe this was possible, and I forgot that it was not really ready for general consumption. Will update as suggested.

Comment (andy-duncan) [inline commands/history.js]: I've pushed these changes to master. I don't think they're super urgent to release, so happy to leave them for the next build.

**Pull request merged**